### ISSUE TITLE
repl: tweaking completion behaviour

### ISSUE THREAD

**Issue by sbinet** [elves/elvish#672]
hi there,

apologies if I missed the obvious place where this is spelled out but...
I was wondering how I could modify the completion of, say, `cd foo/[TAB]`, to not automatically autocomplete with the first item in the list of candidates but just display the candidates.
my muscle-memory expects that only when I hit `[TAB]` a second time I would start to cycle through the candidates.
is there a way to achieve this behaviour?

I looked into https://elvish.io/ref/edit.html#completion-api but nothing jumped at me...

**Comment by zzamboni** [comment]
@sbinet this is how completion in Elvish works so far, and I've also found it hard to get used to. One workaround I've found for this is to enable filter mode automatically in completion mode, by binding the `Tab` key as follows:
```
  edit:insert:binding[Tab] = { edit:completion:smart-start; edit:completion:trigger-filter }
```
That way, when you press tab, the first item will be completed as usual, but filter mode is enabled so you can just keep typing to narrow down the selections.

**Comment by xiaq** [comment]
Yeah, this is how it works (for now). I'm leaving this open as it can be useful to see how to accommodate this use case with a more flexible API.

**Comment by ocehugo** [comment]
IMHO, the solution above should be in the docs - its a huge improvement for newcomers.

PS: I also change edit:completion:binding to use ```&Tab= $edit:insert:start~``` so the behaviour is more familiar. I'm used to use intermittent typing between tab hits to finish some know but long name completions/paths.
